pkg/virt-controller/watch/vm.go

```diff
@@ -217,6 +220,39 @@ func (c *VMController) execute(key string) error {
 		}
 	}
 
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
```

Comment:
I just realized this check is taking place in the VM controller instead of the VMI controller.  doesn't it make sense to catch this at the VMI level?

Reply:
It looked right to me to handle it at the same level as `VMController.handleDataVolumes`.

Reply:
the VM controller has some special behavior that allows someone to both define the VM and DataVolumes in a single manifest. VMIs can reference already created DataVolumes and PersistentVolumeClaims though. 

Here's what I'd expect from this PR.

- The validation webhook should ensure that when a DataVolumeTemplate is defined on a VM, that the resulting volume in the VMI template is referenced as a DataVolume and not a PVC.

- The VMI controller should ensure a referenced PVC isn't a DataVolume before allowing the vmi to be scheduled.

This covers the early detection of a common error during VM creation where a DataVolumeTemplate entry is referenced as a PVC. It also covers the issue where someone can assign a PVC directly to a VMI which could be controlled as a datavolume

Reply:
> * The validation webhook should ensure that when a DataVolumeTemplate is defined on a VM, that the resulting volume in the VMI template is referenced as a DataVolume and not a PVC.

I am not absolutely sure about this one. Most PVC validations happen during creation times of Pods or VMIs and they can also be fixed by modifying the PVC. At the end the PVC can be changed anytime after the VM was created. Also from an ordinay PVC to a PVC which is managed by CDI.

Reply:
@davidvossel By "the validation webhook" do you mean `/virtualmachines-validate` in `virt-api`?

Reply:
> @davidvossel By "the validation webhook" do you mean /virtualmachines-validate in virt-api?

right

> I am not absolutely sure about this one. Most PVC validations happen during creation times of Pods or VMIs and they can also be fixed by modifying the PVC. 

in this case we are defining the DataVolume in the VM spec, so we'd know immediately during POST whether someone is trying to attach their DataVolume as a PVC rather than a DataVolume.  Basically, if the DataVolumeTemplate entry in a VM has the same name as a PVC volume in the VMI template, then that's invalid.

Reply:
Hm, it seems that the validation webhook already catches that situation.

If the spec has the following VM yaml:

```yaml
kind: VirtualMachine
# unimportant
spec:
  # unimportant
  template:
    spec:
     # unimportant
      volumes:
      - name: disk0
        persistentVolumeClaim:
          claimName: test-datavolume
  dataVolumeTemplates:
  - kind: DataVolume
    apiVersion: cdi.kubevirt.io/v1alpha1
    metadata:
      name: test-datavolume
    spec:
      # unimportant
```

Creation returns the following error:
`The request is invalid: spec.dataVolumeTemplate[0]: DataVolumeTemplate entry spec.dataVolumeTemplate[0] must be referenced in the VMI template's 'volumes' list
`